### PR TITLE
Meson: added search for OpenBLAS include directories

### PR DESCRIPTION
**Pull request by ddobbelaere**
Search for OpenBLAS headers in array `get_option('openblas_include')`. Add the first directory that contains `cblas.h` to `includes`.

This fixes following build meson build error on Ubuntu 18.04 with `libopenblas-dev` installed

> meson.build:155:6: ERROR:  Include dir /usr/include/openblas/ does not exist.

### CODE REVIEW

**Comment by borg323** [comment]
What about something like:
`if not cc.has_header('cblas.h')`
`   includes += include_directories(get_option('openblas_include'))`
`endif`

**Comment by ddobbelaere** [comment]
That would also have worked in my specific case (Ubuntu 18.04, as `cblas.h` is in system include directory). The suggested solution also work for multiple possible include directories. It searches for the first match and only adds this directory to `includes`.

**Comment by frpays** [comment]
The proposal of @borg323 is to be examined. If it works as well on every know platform, it's simpler and maybe more in the spirit.

**Comment by borg323** [comment]
My suggestion can be found here: https://github.com/borg323/lc0/tree/openblas_inc

Regarding openblas_dll, it is searched first because the windows build also include a libopenblas.a that is for mingw and doesn't work with MSVS.

File: meson.build [hunk 1]
```diff
@@ -152,7 +152,25 @@ if get_option('build_backends')
 
     elif get_option('openblas') and openblas_lib.found()
       add_project_arguments('-DUSE_OPENBLAS', language : 'cpp')
-      includes += include_directories(get_option('openblas_include'))
+
+      openblas_headers_found = false
```
Comment (mooskagh) [inline meson.build]: That's fine if that's the easiest way to implement that, but maybe there is another way.
I suggest you to show this code in #mesonbuild channel on irc.freenode.org, maybe they will suggest something.
Or I can do that myself tomorrow.

**Comment by frpays** [comment]
Please tell me if got it right.

- In the borg version, 'openblas_include' is used only if the blas header is not found in standard directories. If this case, it has to exist, and contain the blas headers otherwise the build fails.

- In the original PR version, you could actually fill the 'openblas_include' with all the possible paths when you can find the openblas headers on all platform. Then the for loop will chose the first one containing the blas headers. If non contains the blas headers, the build fails.

borg's version is simpler but asks some users to do some research on fix the meson_options.
orignal version has the advantage that once you gathered all possible paths, there's no need for anyone to change this part on meson_options. It's more "magical".

If I not mistaking, then I can of like the original version.

**Comment by ddobbelaere** [comment]
You are right, this was my intention from the beginning (look for the first valid include directory and only add this to includes, in line with issue https://github.com/LeelaChessZero/lc0/issues/112). This gives us the flexibility to add an extra include directory to the option `openblas_include`. I'm glad you like it.

I like @borg323's suggestion to look for the presence of `openblas_config.h`, I've changed this. I also tried to make the code a bit more compact. Sorry for the commit mess, I am a bit new to pull requests, but it should be alright now.

**Pull request merged**